File: ProblemSet2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>

int main(void)
{
    int countw = 0, i;
    int counts = 0, j;
    int countl = 0, k;
    char p[1000];

//prompt the text from user
    printf("Text: ");
    scanf("%[^\n]", p);

//counting number of words
    for (i = 0 ; p[i] != '\0'; i++)
    {
        if (p[i] == ' ' && p[i + 1] != ' ')

        countw++;

    }

//counting number of sentences
    for (j = 0 ; p[j] != '\0'; j++)
    {
        if (p[j] == '.' || p[j] == '?' || p[j] == '!')

        counts++;

    }

//counting number of letters
    for (k = 0 ; p[k] != '\0'; k++)
    {
        if ((p[k] >= 'a' && p[k] <= 'z') || (p[k] >= 'A' && p[k] <= 'Z'))

        countl++;

    }

//counting number of letters per 100 words
    float L = ((float)countl / (countw + 1)) * 100;

//counting number of sentences per 100 words
    float S = ((float)counts / (countw + 1)) * 100;

//readability
    int R = round((0.0588 * L) - (0.296 * S) - 15.8);

//grade
    if (R < 1)
    printf("Before Grade 1\n");

    else if (R >= 1 && R <= 16)
    printf("Grade %i\n", R);

    else
    printf("Grade 16+\n");
}

File: ProblemSet6/readability/readability.py
from cs50 import get_string
import math
#inspired from try0read

#Coleman-Liau index = 0.0588 * L - 0.296 * S - 15.8
#L = number of letters per 100 words
#S = number of sentences per 100 words
while True:
    p = get_string("Text: ")

    l = 0 #letters
    s = 0 #sentences
    w = 1 #words
    h = 0 #dummy

    if(str):
        for i in range(len(p)):
            if (p[i].isalpha()):
                l += 1
            elif (p[i].isnumeric()):
                l += 1
            elif (p[i] == ' ' and p[i + 1] != ' '):
                w += 1
            elif (p[i] == '.' or p[i] == '?' or p[i] == '!'):
                s += 1
            else:
                h += 1

        L = float ((l / w) * 100)
        S = float ((s / w) * 100)

#readability
        R = round((float)(0.0588 * L - 0.296 * S - 15.8))

        if (R >= 16):
            print('Grade 16+')
        elif (R < 1):
            print('Before Grade 1')
        else:
            print('Grade ', R)

        break